index.js

```diff
@@ -7,12 +7,174 @@ function ask(questionText) {
+    game = await ask(`What version of guess the number would you like to play?`);
+    game = +game;
+  }
+  game = +game;
+  min = await ask(`Now, what should we set for the minimum value (zero or greater)?`);
+  min = +min;
+  while (Number.isInteger(min) !== true || min < 0) {
+    console.log(`Please enter a positive integer of zero or greater.`);
+    min = await ask(`What would you like to set for the minimum?`);
+    min = +min;
+  }
+  max = await ask(`What would you like for the maximum number to be?`);
+  max = +max;
+  while (Number.isInteger(max) !== true || max <= min) {
+    console.log(`Please enter an integer greater than ${min}.`);
+    max = await ask(`Please select a maximum value.`);
+    max = +max;
+  }
+  if (game === 1) {
+    computerGuess = split(min, max);
+    computerGuess = Math.floor(computerGuess);
+    //asks for the secret number and converts it to integer.
+    secretNumber = await ask("What is your secret number? I won't peek, I promise...\n");
+    secretNumber = +secretNumber;
+    //error checking
+    while (Number.isInteger(secretNumber) !== true || secretNumber < min || secretNumber > max || secretNumber === 0) {
+      console.log(`Please enter a whole number that's greater than ${min} and less than ${max}.`);
+      secretNumber = await ask(`What would you like your secret number to be?`);
+      secretNumber = +secretNumber;
+    }
+    guess();
+  }
+  if (game === 2) {
+    console.log(`We'll get to this later.`);
+    process.exit();
+  }
 }
+
+async function guess() {
+  let answer = await ask(`Is your number... ${computerGuess}?(Yes or no)\n>_`);
+  count = count + 1;
+  //we wait to sanitize input only once they have evtered a value, because sanitize cannot lowercase undefined.
+  while(answer === ``) {
+    console.log(`Please enter yes or no.`);
+    answer = await ask(`Is your number... ${computerGuess}?\n>_`);
+  }
+  sanitize(answer);
+  //makes sure user input 
+  //while it is not the right guess
+    while (ANSWERS_POS.includes(answer)) {
+      if (secretNumber !== computerGuess) {
+      console.log(`Are you sure about that?`);
+      answer = await ask(`Is your number... ${computerGuess}?\n>_`);
+      sanitize(answer);
+    } if (secretNumber === computerGuess) {
+      computerScore = computerScore + 1;
+      return youWin();
+    }
+  }
+    while (ANSWERS_NEG.includes(answer)) {
+      if (secretNumber !== computerGuess) {
+        return higherLower();
+      }
+      if (secretNumber === computerGuess) {
+        console.log(`Please enter yes or no.`);
+        answer = await ask(`Is your number... ${computerGuess}?\n>_`);
+        sanitize(answer);
+    }
+  }
+}
+
+async function higherLower() {
+  let higher = await ask(`Is it higher (h) or lower (l) than ${computerGuess}?\n>_`);
+  while (higher === ``) {
+    console.log(`Please say if it's higher or lower.`);
+    higher = await ask(`Is it higher or lower than ${computerGuess}?`);
+  }
+  sanitize(higher);
```

Comment:
Same as in the guess function, currently the game will only accept values in the array, and they will not be sanitized at the moment.  This can be fixed in the same way as above.  Since you are calling this while loop in multiple places, you could pull it out and make the sanitize function a little more robust, and be in charge of all of the error handling.  This would save you from needing to check for the empty string and call sanitize on each input.